src/PIL/IcnsImagePlugin.py

```diff
@@ -302,65 +300,70 @@ def load(self):
         self.load_end()
 
 
+def _to_int(s):
+    b = s.encode("ascii")
+    return (b[0] << 24) | (b[1] << 16) | (b[2] << 8) | b[3]
+
+
+MAGIC = _to_int("icns")
+TOC = "TOC "
+
+
 def _save(im, fp, filename):
     """
     Saves the image as a series of PNG files,
     that are then converted to a .icns file
-    using the macOS command line utility 'iconutil'.
 
-    macOS only.
+    Support for arbitrary systems
     """
     if hasattr(fp, "flush"):
         fp.flush()
 
-    # create the temporary set of pngs
-    with tempfile.TemporaryDirectory(".iconset") as iconset:
-        provided_images = {
-            im.width: im for im in im.encoderinfo.get("append_images", [])
-        }
-        last_w = None
-        second_path = None
-        for w in [16, 32, 128, 256, 512]:
-            prefix = "icon_{}x{}".format(w, w)
-
-            first_path = os.path.join(iconset, prefix + ".png")
-            if last_w == w:
-                shutil.copyfile(second_path, first_path)
-            else:
-                im_w = provided_images.get(w, im.resize((w, w), Image.LANCZOS))
-                im_w.save(first_path)
-
-            second_path = os.path.join(iconset, prefix + "@2x.png")
-            im_w2 = provided_images.get(w * 2, im.resize((w * 2, w * 2), Image.LANCZOS))
-            im_w2.save(second_path)
-            last_w = w * 2
-
-        # iconutil -c icns -o {} {}
-
-        convert_cmd = ["iconutil", "-c", "icns", "-o", filename, iconset]
-        convert_proc = subprocess.Popen(
-            convert_cmd, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL
+    # size
```

Comment:
For consistency with `# Header`, etc
```suggestion
    # Size
```